state/pgstatestorage/pgstatestorage.go

```diff
@@ -363,6 +363,25 @@ func (p *PostgresStorage) GetBatchL2DataByNumber(ctx context.Context, batchNumbe
 	var batchL2Data []byte
 	err := q.QueryRow(ctx, getBatchL2DataByBatchNumber, batchNumber).Scan(&batchL2Data)
 
+	if errors.Is(err, pgx.ErrNoRows) {
```

Comment:
This is more readable to me:

```suggestion
	if err != nil {
	    if errors.Is(err, pgx.ErrNoRows) {
	         return p.GetBatchL2DataByNumberFromBackup(ctx, batchNumber, dbTx)
	    }
	    
	    return nil, err
	}
```